mlflow/store/s3_artifact_repo.py

```diff
@@ -28,14 +29,15 @@ def log_artifact(self, local_file, artifact_path=None):
         if artifact_path:
             dest_path = build_path(dest_path, artifact_path)
         dest_path = build_path(dest_path, os.path.basename(local_file))
-
-        boto3.client('s3').upload_file(local_file, bucket, dest_path)
+        s3_endpoint_url = os.environ.get('MLFLOW_S3_ENDPOINT_URL')
+        boto3.client('s3', endpoint_url=s3_endpoint_url).upload_file(local_file, bucket, dest_path)
 
     def log_artifacts(self, local_dir, artifact_path=None):
         (bucket, dest_path) = data.parse_s3_uri(self.artifact_uri)
         if artifact_path:
             dest_path = build_path(dest_path, artifact_path)
-        s3 = boto3.client('s3')
+        s3_endpoint_url = os.environ.get('MLFLOW_S3_ENDPOINT_URL')
```

Comment:
Could we pull this logic into an internal `_get_s3_client(self)` method so that we don't need to remember to check the environment variable whenever we need an S3 client (e.g. if we add a new ArtifactRepository method)?

Reply:
thank you it is a good point, I will do it